### DISCUSSION TITLE
Trying to understand relation and expand

### DISCUSSION THREAD

**Discussion by Iccr** [pocketbase/pocketbase#6199] [Q&A]
Hi, I am trying to understand the "relation" and "expand"  behaviour in pocketbase, 

I am trying to do 1 to many relation. I have province that can have many districts.

<img width="668" alt="Screenshot 2024-12-28 at 8 18 27 PM" src="https://github.com/user-attachments/assets/43e5061e-c891-450d-adfd-312762b850d5" />

from client side I am fetching it like this

    
```
final records =
          await client.collection("provinces").getFullList(expand: "districts");

```
and  this is the response I get.
<img width="627" alt="Screenshot 2024-12-28 at 8 16 21 PM" src="https://github.com/user-attachments/assets/ffff8c0b-867c-4793-9d1b-e36c6caa1c15" />

I can see that The districts is there, but instead of object I can only see the id.

from the proxyman, request is
`https://something.pockethost.io/api/collections/provinces/records?page=1&perPage=500&expand=districts&skipTotal=true
`

and response is

```
{
  "items": [
    {
      "collectionId": "pbc_4091735620",
      "collectionName": "provinces",
      "created": "2024-12-28 19:43:05.776Z",
      "districts": [
        "4lwk75tfemxg56n",
        "7643ppp2p61070d",
        "1ogir7r859a2e32"
      ],
      "id": "pp44v1z531l8w8k",
      nepali_title": "other tile",
      "title": "Province 1",
      "updated": "2024-12-28 20:42:39.010Z"
    }
  ],
  "page": 1,
  "perPage": 500,
  "totalItems": -1,
  "totalPages": -1
}
```

Also I am also getting same response without the expand keyword. Am I doing something wrong.

I was expecting with expand keyword, I would be getting the object in the districts instead of the ids. Am I doing something wrong?

**Comment by Iccr** [comment]
I figured out , the issue was with the view api rule. after editing the view api rule for district and province, finally i am able to get this working.

**Reply by 100wattapp** [reply]
Hi, sorry to bother you, I have the same problem, and for a few days I can't get out of it.
I completely deleted the filters, but nothing in the relationship field I get back only the id...
can you help me? thanks

**Reply by Iccr** [reply]
> Hi, sorry to bother you, I have the same problem, and for a few days I can't get out of it. I completely deleted the filters, but nothing in the relationship field I get back only the id... can you help me? thanks

In the pocketbase, in the db Colection which you want to expand, edit the collection, see the api rules section, there the  click on the view rule and keep it empty. After making that changes, I was able to get the object in the expanded fields.

<img width="667" alt="Screenshot 2025-01-06 at 11 38 38 AM" src="https://github.com/user-attachments/assets/538fae62-a63c-4f45-a67d-9d717882aff8" />

**Comment by 100wattapp** [comment]
Hi, there was something not updated in pocketbase, because, I had already deactivated all the viewing permissions, but the situation was unblocked when I put the permissions back and I deleted them again.
Thanks for the suggestion.
